testsuite/tree_sequences/test_generate_data_matrix.cc

```diff
@@ -0,0 +1,42 @@
+#include <iostream>
+#include <algorithm>
+#include <boost/test/unit_test.hpp>
+#include <fwdpp/ts/marginal_tree_functions/roots.hpp>
+#include <fwdpp/ts/marginal_tree_functions/nodes.hpp>
+#include <fwdpp/ts/generate_data_matrix.hpp>
+#include "simple_table_collection_infinite_sites.hpp"
+
+BOOST_AUTO_TEST_SUITE(test_generate_data_matrix)
```

Comment:
There are other safety tests to think about:

1. Two mutations at the same site must not have the same key.  Not super-easy to test as they could be out-of-order.
2. All mutations at a site should have the same "neutral" flag. Maybe?  Maybe not?